library/Icinga/Crypt/AesCrypt.php

```diff
@@ -51,21 +58,15 @@ class AesCrypt
     private $tag;
 
     /** @var string The cipher method */
-    private $method = 'AES-128-GCM';
-
-    const GCM_SUPPORT_VERSION = '7.1';
+    private $method;
 
     public function __construct($random_bytes_len = 128)
     {
-        if (version_compare(PHP_VERSION, self::GCM_SUPPORT_VERSION, '<')) {
-            $this->method = 'AES-128-CBC';
-        }
-
         $this->key = random_bytes($random_bytes_len);
```

Comment:
While you're at it lightening the constructor... :smile: , why not also generating the key lazily? `getKey()` also throws in case of a missing key, which never will be the case since it's set up here in the constructor. Generating the key in `getKey()` if there's none is then a clear possibility to me. (Of course, you'd need a property for the key length then, but so what?)